android_app/app/src/main/java/com/health/openscale/gui/MainActivity.java

```diff
@@ -651,10 +657,92 @@ public void handleMessage(Message msg) {
                         Timber.e("Bluetooth scale message error: " + ex);
                     }
                     break;
+                case CHOOSE_SCALE_USER:
+                    chooseScaleUser(msg);
+                    break;
+                case ENTER_SCALE_USER_CONSENT:
+                    enterScaleUserConsent(msg);
+                    break;
             }
         }
     };
 
+    private void chooseScaleUser(Message msg) {
+        AlertDialog.Builder mBuilder = new AlertDialog.Builder(MainActivity.this);
+        Pair<CharSequence[], int[]> choices = (Pair<CharSequence[], int[]>)msg.obj;
+
+        mBuilder.setTitle(getResources().getString(R.string.info_select_scale_user));
+        mBuilder.setSingleChoiceItems(choices.first, -1 , new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialoginterface, int i) {
+                    Timber.d("UI selected " + i + ": " + choices.first[i] + " P-0" + choices.second[i]);
+                    OpenScale.getInstance().setBluetoothDeviceUserIndex(OpenScale.getInstance().getSelectedScaleUser().getId(), choices.second[i], callbackBtHandler);
```

Comment:
Maybe we could set directly the scale user index  preference here.

Reply:
I wanted to re-initiate connection to scale after choice has been made. Therefore i must signal to BluetoothStandardWeightProfile that user made a choice and we may restart state machine re-connect using user's input.

And I did not want to change properties/settings that are private to BluetoothStandardWeightProfile from GUI.